File: pset2/caesar/caesar.c
#include<string.h>
#include<cs50.h>
#include<stdlib.h>
#include<stdio.h>
#include<ctype.h>

int n = 0, i = 0, c = 0;
string a;

int main(int argc, string argv[])
{
    if(argc != 2)                                          //Checking argument count for 2
    {
        printf("Usage: %s k\n", argv[0]);
        return 1;
    }
    int k = atoi(argv[1]);                                 //Converting string argv[1] to integer
    int e = (k % 26);                                      //Obtaining Key for shifting plaintext
    printf("plaintext: ");
    a = get_string();                                      //Getting input from user
    printf("ciphertext: ");
    for(i = 0, n = strlen(a); i < n; i++)
    {
        if(isalpha(a[i]))                                  //Checking if the elemnets entered by user are alphabet
        {
            if((a[i] >= 'a') && (a[i] <= 'z'))
            {
                a[i] = toupper(a[i]);                      //To avoid the condition of overflow in signed char
                a[i] += e;                                 //Shifting plaintext by desired amount
                if(a[i] > 90)                              //Condition to wrap from Z to A
                {
                    int d = a[i] - 90;
                    a[i] = 65 + d - 1;
                }
                a[i] = tolower(a[i]);                      //Converting uppercase characters to lowercase characters
                printf("%c", a[i]);                        //Printing Output
            }
            if((a[i] >= 'A') && (a[i] <= 'Z'))
            {
                a[i] += e;                                 //Shifting plaintext by desired amount
                if(a[i] > 90)                              //Condition to wrap from Z to A
                {
                    int d = a[i] - 90;
                    a[i] = 65 + d - 1;
                }
                printf("%c", a[i]);                        //Printing Output
            }
            
        }
        else                                               //If the element is non - alphabetical
            printf("%c", a[i]);                            //Printing it as it is
    }
    printf("\n");
    return 0;
}

File: pset6/caesar.py
import cs50
import sys

b = []
if len(sys.argv) != 2:
    print("Usage: {} k".format(sys.argv[0]))

k = int(sys.argv[1])
e = k % 26
a = cs50.get_string("plaintext: ")
print("ciphertext: ", end="")
for i in range(0, len(a)):
    if a[i].isalpha() == True:
        if a[i] >= 'a' and a[i] <= 'z':
            b.append(chr(ord(a[i]) + e))
            if ord(b[i]) > 122:
                d = ord(b[i]) - 122
                b[i] = 97 + d - 1
            print("%c" %(b[i]), end="")
        if a[i] >= 'A' and a[i] <= 'Z':
            b.append(chr(ord(a[i]) + e))
            if ord(b[i]) > 90:
                d = ord(b[i]) - 90
                b[i] = 65 + d - 1
            print("%c" %(b[i]), end="")
    else:
        print("{}".format(chr(b[i]), end=""))
print()